common/app/common/Logging.scala

```diff
@@ -25,6 +29,26 @@ trait Logging {
   def logErrorWithCustomFields(message: String, error: Throwable, customFields: List[LogField]): Unit = {
     log.logger.error(customFieldMarkers(customFields), message, error)
   }
+
+  def errorLoggingF[A](message: String)(task: => Future[A])(implicit ec: ExecutionContext): Future[A] = {
+    def errorLoggingFuture(f: Future[A]): Future[A] = {
+      f.recover {
+        case NonFatal(e) => Logger.error(message, e) ; throw e
+      }
+    }
+
+    Try(task) match {
+      case Success(f) => errorLoggingFuture(f)
+      case Failure(NonFatal(e)) => Logger.error(message, e) ; throw e
```

Comment:
Try only ever catches NonFatal anyway, so the Failure would always contain one of those

Reply:
Thanks for the tip!